### PR TITLE
http code semantizat

### PR DESCRIPTION
**Pull request by rokkiter**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Make http code semantic
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/client-go/util/certificate/certificate_manager_test.go [hunk 1]
```diff
@@ -963,7 +964,7 @@ func TestServerHealth(t *testing.T) {
 			certs:       currentCerts,
 
 			failureType:      createError,
-			clientErr:        apierrors.NewGenericServerResponse(404, "POST", schema.GroupResource{}, "", "", 0, true),
+			clientErr:        apierrors.NewGenericServerResponse(http.StatusUnauthorized, "POST", schema.GroupResource{}, "", "", 0, true),
```
Comment (MikeSpreitzer) [inline staging/src/k8s.io/client-go/util/certificate/certificate_manager_test.go:967]: This should be StatusNotFound, right?

File: staging/src/k8s.io/client-go/util/certificate/certificate_manager_test.go [hunk 2]
```diff
@@ -972,7 +973,7 @@ func TestServerHealth(t *testing.T) {
 			certs:       currentCerts,
 
 			failureType:      createError,
-			clientErr:        apierrors.NewGenericServerResponse(404, "POST", schema.GroupResource{}, "", "", 0, false),
+			clientErr:        apierrors.NewGenericServerResponse(http.StatusUnauthorized, "POST", schema.GroupResource{}, "", "", 0, false),
```
Comment (MikeSpreitzer) [inline staging/src/k8s.io/client-go/util/certificate/certificate_manager_test.go:976]: This should be StatusNotFound, right?

**Review by MikeSpreitzer — CHANGES_REQUESTED** [review]
A couple of copy-and-paste errors noted inline.

**Comment by MikeSpreitzer** [comment]
@rokkiter : would you like to follow up here?

**Comment by MikeSpreitzer** [comment]
@rokkiter : thank you for this submission.  I think it just needs a couple of corrections, and it needs the release note filled in (I think NONE is a fine answer there).

File: staging/src/k8s.io/client-go/discovery/cached/memory/memcache.go [hunk 3]
```diff
@@ -79,7 +80,7 @@ func isTransientError(err error) bool {
 		return true
 	}
 
-	if t, ok := err.(errorsutil.APIStatus); ok && t.Status().Code >= 500 {
+	if t, ok := err.(errorsutil.APIStatus); ok && t.Status().Code >= http.StatusInternalServerError {
```
Comment (sttts) [inline staging/src/k8s.io/client-go/discovery/cached/memory/memcache.go:83]: these`>= constant` cases IMO make the code harder to read. Would revert them. `== constant` is fine.

**Pull request closed without merging**